### PR TITLE
fix: add optional chaining back to `finalise` hook

### LINKED ISSUE #8841: [bug] Build errors swallowed by `finalise is not a function`
### Describe the bug

After the latest update, all build-time errors are swallowed and replaced with:

```rust
[vite-plugin-sveltekit-compile] finalise is not a function
error during build:
TypeError: finalise is not a function
```

Seems to be coming from: https://github.com/sveltejs/kit/blame/089c405041b0267356281fc8ea5f19ecc331178b/packages/kit/src/exports/vite/index.js#L728

### Reproduction

I just changed an import path in `+page.svelte` to a non-existent one (`./ASDF/Counter.svelte`) to repro:

https://stackblitz.com/edit/sveltejs-kit-template-default-xnumuj?file=src/routes/+page.svelte

### Logs

```Shell
✓ 141 modules transformed.
transforming (145) src/styles/prism.css4:11:11 PM [vite-plugin-svelte] ssr compile done.
package                 files    time      avg
@REDACTED/REDACTED       40   0.53s   13.3ms
[vite-plugin-sveltekit-compile] finalise is not a function
error during build:
TypeError: finalise is not a function
    at Object.handler (file:///Users/REDACTED/dev/Projects/REDACTED/REDACTED/REDACTED/node_modules/.pnpm/@sveltejs+kit@1.3.7_svelte@3.55.1+vite@4.0.4/node_modules/@sveltejs/kit/src/exports/vite/index.js:722:11)
```

### System Info

```Shell
System:
    OS: macOS 13.2
    CPU: (16) x64 Intel(R) Core(TM) i9-9980HK CPU @ 2.40GHz
    Memory: 27.92 GB / 64.00 GB
    Shell: 5.8.1 - /bin/zsh
  Binaries:
    Node: 16.19.0 - ~/.volta/tools/image/node/16.19.0/bin/node
    Yarn: 1.22.18 - ~/.volta/tools/image/yarn/1.22.18/bin/yarn
    npm: 8.19.3 - ~/.volta/tools/image/node/16.19.0/bin/npm
  Browsers:
    Brave Browser: 109.1.47.186
    Chrome: 111.0.5562.0
    Chrome Canary: 112.0.5572.1
    Edge: 83.0.478.44
    Firefox: 89.0
    Firefox Developer Edition: 106.0
    Firefox Nightly: 76.0a1
    Safari: 16.3
    Safari Technology Preview: 16.4
  npmPackages:
    @sveltejs/adapter-vercel: 1.0.6 => 1.0.6
    @sveltejs/kit: 1.3.7 => 1.3.7
    @sveltejs/package: 1.0.2 => 1.0.2
    svelte: 3.55.1 => 3.55.1
    vite: 4.0.4 => 4.0.4
```

### Severity

serious, but I can work around it

### Additional Information

_No response_

### PR DESCRIPTION
**Pull request by FractalHQ**
### Please don't delete this checklist! Before submitting the PR, please make sure you do the following:
- [x] It's really useful if your PR references an issue where it is discussed ahead of time. In many cases, features are absent for a reason. For large changes, please create an RFC: https://github.com/sveltejs/rfcs
- [x] This message body should clearly illustrate what problems it solves.
- [ ] Ideally, include a test that fails without this PR but passes with it.

### Tests
- [ ] Run the tests with `pnpm test` and lint the project with `pnpm lint` and `pnpm check`

### Changesets
- [x] If your PR makes a change that should be noted in one or more packages' changelogs, generate a changeset by running `pnpm changeset` and following the prompts. Changesets that add features should be `minor` and those that fix bugs should be `patch`. Please prefix changeset messages with `feat:`, `fix:`, or `chore:`.

fixes #8841

### CODE REVIEW

**Comment by benmccann** [comment]
is this something we could add a test for? If it only occurs during build errors it could be added in `packages/kit/test/build-errors`

**Comment by Rich-Harris** [comment]
added a test

File: .changeset/grumpy-islands-collect.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@sveltejs/kit': patch
+---
+
+fix: add optional chaining back to `finalise` hook
```
Comment (Conduitry) [inline .changeset/grumpy-islands-collect.md:5]: Can we describe this in a more user-friendly way?

Comment (Rich-Harris) [reply .changeset/grumpy-islands-collect.md:5]:
```suggestion
fix: preserve build error messages
```

Comment (Conduitry) [reply .changeset/grumpy-islands-collect.md:5]: Sounds good to me.

**Review by Rich-Harris — APPROVED** [review]

**Comment by Rich-Harris** [comment]
thanks!

**Pull request merged**